docs/decisions/0014-single-taxonomy-view-api.rst

```diff
@@ -0,0 +1,126 @@
+13. Single taxonomy view API
+=====================================
+
+
+Context
+--------
+
+This view returns tags of a closed taxonomy (for MVP has not been implemented yet
+for open taxonomies). It is necessary to make a decision about what structure the tags are going 
+to have and how the pagination is going to work. It was taken into account that a taxonomy can
+have a large number of tags and are mostly represented as trees.
+
+**Branch:** Is the representation of a root Tag and all its children up to the leaves.
+
+
+Decision
+---------
+
+
+Tag representation
+~~~~~~~~
+
+Return a list of root tags and within each one have the list of children tags. Each root would have
+its entire branch. The list of root tags will be ordered alphabetically as is each listing at each level of the tree.
+
+{
+    "count": 100,
+    "tags": [
+        {
+            "id": "tag_1",
+            "value": "Tag 1",
+            "taxonomy_id": "1",
+            "sub_tags": [
+                {
+                    "id": "tag_2",
+                    "value": "Tag 2",
+                    "taxonomy_id": "1",
+                    "sub_tags": [
+                        (....)
+                    ]
+                },
+                (....)
+            ]
+        },
+        (....)
+    ]
+}
+
+
+**Pros:**
+
+- The edX's interfaces show the tags in the form of a tree.
+- The frontend needs no further processing as it is in a displayable format.
+- It is kept as a simple implementation.
+
+**Cons:**
+
+- More implementation on the API side.
+
+
+Pagination
+~~~~~~~~~~~
+
+Apply the pagination only in the root tags and bring the entire branch of each root.
+Children do not affect pagination in any way.
+
+**Pros**
+
+- It is the simplest way.
+
+**Cons**
+
+- The children would not have pagination, in the long run there may be cases in which
+  the branch has hundreds of children, and they would still all be brought.
+
```

Comment:
I agree with your chosen option @ChrisChV , that makes total sense.

There's two more use cases implied by the UI:

![tag_search_sort](https://github.com/openedx/openedx-learning/assets/7556571/89d07762-3284-43ac-b04a-8f9dadbb18d8)

**Sort A-Z or Z-A**

This sorts the top-level tags, not the child tags underneath (cf [figma comment](https://www.figma.com/file/DmUC2e7BAoQsGDq4KgApPa?node-id=1237:117997&mode=design#504230063)).

**Search for tags in a taxonomy**

This one's a little more complicated since it could be done a couple of ways.

Options I see:

1. We support tag search on the backend, and return a subset of matching tags in the format proposed here.
  It means a backend API hit every time someone searches for a tag, but will scale.
1. We constrain the number of tags allowed in a taxonomy for MVP, so that the API can return all the tags in one page.
  This allows the frontend to handle tag search all on its own, which is performant, but doesn't scale.

@ChrisChV  and @bradenmacdonald  How do you think we should handle this?

Reply:
@bradenmacdonald Yeah, I don't think there's an upper bound planned for post-MVP, and there's already [huge taxonomies being discussed](https://openedx.slack.com/archives/C03T0RZEASG/p1692043507665879), as you noted. So I agree: using the backend for search is the safest (and simplest) option.

Reply:
Considering [taxonomies like this](https://openedx.slack.com/archives/C03T0RZEASG/p1692043507665879), I also agree with the search option in the backend. But another concern arises, in the [interface to tag an object](https://github.com/openedx/openedx-learning/pull/59#discussion_r1274400990), will all the tags be brought in as initially thought? I think this page should use this same view (with pagination) and the same search

Reply:
@ChrisChV Given the huge taxonomies shown, I think we'll have to find a different approach for that page. We can't bring in all the tags as we initially thought. It would be too big.

Reply:
I was thinking of using the same view for both screens, since the screen to tag an object must also be paginated, I'm going to [ask](https://www.figma.com/file/DmUC2e7BAoQsGDq4KgApPa?node-id=1515:84066&mode=design#535383862) UX about this